### PR TITLE
Va cowamm banner

### PR DESCRIPTION
**Pull request by fairlighteth**
# Summary

- Creates a unified CoW AMM banner component that can serve different variations depending on location of usage. For now `global` and `tokenSelector` are the two available locations.
- DEMO only dropdown activated for this PR
- Includes click analytics events

https://github.com/user-attachments/assets/318aba88-c30d-4565-a009-95426da4a112

https://github.com/user-attachments/assets/61e39640-7800-4a9c-be4d-795a3c77a955

### CODE REVIEW

**Comment by fairlighteth** [comment]
@elena-zh Did some updates:
- Don't show banners when not connected with a wallet
- Added an extra scenario:

Screenshot 2024-10-16 at 12 05 49

**Comment by elena-zh** [comment]
Hey @fairlighteth , great job!
 
Some cases/issues:
1. Safe or any other SC-wallet: it should not say that transactions are gas-free there
image
image

2. Mobile, real devices: tokens info is not displayed on a banner inside token list: 
image
3. Colors for banners are differ in the light and dark modes. Is it done intentionally?
dark
light
4. I think, it would be great to add one more case when a user is connected to an unsupported network. IMO, it should behave the same as to a not connected to a wallet?
5. (sorry if I've missed this somewhere else) What should we show when there is no pool in a current network (e.g. Sepolia)? OR no CoW AMM pools in a selected network (again, on Sepolia, but in the future in Base, for example)?

Thanks!

**Comment by fairlighteth** [comment]
@elena-zh thank you for reviewing this:
1. Fixed
2. Fixed
3. Done intentionally
4. Yes correct, fixed now
5. Balancer supports these networks, which we all cover (except for Sepolia, but that's just for testing anyway...)
Screenshot 2024-10-18 at 14 24 25

So in reference to point 4, right now we only show the banners for networks we support in our UI.

**Comment by elena-zh** [comment]
@fairlighteth , thank you!

To clarify the 5th case I reported above: 
We don't have CoW AMM pools on Sepolia on Prod (however, it is a supported network). 
Same as for other sources. 
My question was more about what message should we show on the c=CoW AMM banner in this case?

**Comment by fairlighteth** [comment]
@elena-zh for now I think we show the general message, which would link you to CoW AMM balancer's page. For now I wouldn't make the message to specific saying "Switch to Ethereum and explore pools" etc.

**Review by elena-zh — APPROVED** [review]
Approved then! :)

File: apps/cowswap-frontend/src/common/pure/CoWAMMBanner/CoWAmmBannerContent.tsx [hunk 1]
```diff
@@ -0,0 +1,396 @@
...
+  isDemo,
+  selectedState,
+  setSelectedState,
+  dummyData,
+  lpTokenConfig,
+  onCtaClick,
+  onClose,
+}: CoWAmmBannerContentProps) {
+  const isMobile = useMediaQuery(upToSmall)
+  const isDarkMode = useIsDarkMode()
```
Comment (alfetopito) [inline apps/cowswap-frontend/src/common/pure/CoWAMMBanner/CoWAmmBannerContent.tsx:83]: Ideally these should come as props, given this is a pure component

File: apps/cowswap-frontend/src/modules/tokensList/pure/TokensVirtualList/index.tsx [hunk 2]
```diff
@@ -33,7 +38,11 @@ export function TokensVirtualList(props: TokensVirtualListProps) {
     props
   const { values: balances, isLoading: balancesLoading } = balancesState
 
+  const { account: connectedAccount } = useWalletInfo()
+
+  const isInjectedWidgetMode = isInjectedWidget()
   const isWalletConnected = !!account
+  const isChainIdUnsupported = useIsProviderNetworkUnsupported()
```
Comment (alfetopito) [inline apps/cowswap-frontend/src/modules/tokensList/pure/TokensVirtualList/index.tsx:45]: Same thing, pure component, don't use hooks

File: apps/cowswap-frontend/src/common/pure/CoWAMMBanner/CoWAmmBannerContent.tsx (hunk 1 above)
Comment (shoom3301) [reply apps/cowswap-frontend/src/common/pure/CoWAMMBanner/CoWAmmBannerContent.tsx:83]: I will refactor that part along with connecting it to the logic

File: apps/cowswap-frontend/src/modules/tokensList/pure/TokensVirtualList/index.tsx (hunk 2 above)
Comment (shoom3301) [reply apps/cowswap-frontend/src/modules/tokensList/pure/TokensVirtualList/index.tsx]: I will refactor that part along with connecting it to the logic

**Review by shoom3301 — APPROVED** [review]

**Pull request merged**